### DISCUSSION TITLE
arbitrary counters

### DISCUSSION THREAD

**Discussion by naidoo88** [dglazier/chanser#28] [User problems]
Hi all, 

I want to include some arbitrary counters to have some general stats on my code (eg. how many events have photons detected somewhere other than the FD or FT?; how many dubious photons hit the ECAL while depositing no energy in the PCAL?).

What is the best way to do this in chanser?  Specifically, where would you initialise a counter with zero such that it's not reset every event? 

It would likely be sufficient to simply print these to screen, but out of interest is it possible to add them to the FINALOUTTREE for future reference?

**Comment by dglazier** [comment]
Chanser writes events for every combitorial of every event that passes cuts. So you need to decide  it this is when you want to increment your counters.
If you include the counter in your tree data initialised to 0 then it will not reset every event unless you explicitly do so.
For counters output to screen you may just include those as datamembers of your class.
If you just want to count based on actual tracks in the event rather than combitorial events, then this might be more suited to a detected Mask class that is called once per event, rather than once per combitorial.
Or if you think it is useful we could add a FinalState::UserInitEvent()  which is called once per read event.
